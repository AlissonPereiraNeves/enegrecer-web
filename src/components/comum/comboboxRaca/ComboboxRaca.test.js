import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ComboboxRaca from '../comboboxRaca';
import TIPO from '../comboboxRaca/tipos';

describe('Combobox Raça', () => {
  it('deve possuir estrutura definida quando tipo igual testemunha ', () => {
    const onChange = jest.fn();
    const comboboxRacaTestemunha = mount(<ComboboxRaca id={'raca-testemunha'} onChange={onChange} classes={'racas'} tipo={TIPO.TESTEMUNHA} />);
    const comboboxRacaTestemunhaTextoJson = toJson(comboboxRacaTestemunha);
    expect(comboboxRacaTestemunhaTextoJson).toMatchSnapshot();
  });

  it('deve possuir estrutura definida quando tipo igual vítima ', () => {
    const onChange = jest.fn();
    const comboboxRacaVitima = mount(<ComboboxRaca id={'raca-vitma'} onChange={onChange} classes={'racas'} tipo={TIPO.VITIMA} />);
    const comboboxRacaVitimaTextoJson = toJson(comboboxRacaVitima);
    expect(comboboxRacaVitimaTextoJson).toMatchSnapshot();
  });
});
