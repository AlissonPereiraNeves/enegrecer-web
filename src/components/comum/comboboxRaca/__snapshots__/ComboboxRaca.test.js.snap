// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Combobox Raça deve conter somente as raças Pardo e Negro quando somenteRacasVitima é true  1`] = `
<combobox
  divClasse="racas"
  id="raca-vitma"
  itens={
    Array [
      "Preta",
      "Parda",
    ]
  }
  label="Selecione a Raça:"
  state="raca-testemunha"
  valorPadrao="Selecione a Raça:"
/>
`;

exports[`Combobox Raça deve conter todas as raças quando somenteRacasVitima é false  1`] = `
<combobox
  divClasse="racas"
  id="raca-testemunha"
  itens={
    Array [
      "Amarela",
      "Branco",
      "Índigena",
      "Parda",
      "Preta",
    ]
  }
  label="Selecione a Raça:"
  state="raca-testemunha"
  valorPadrao="Selecione a Raça:"
/>
`;
